uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/BaseEditor.java

```diff
@@ -317,107 +324,134 @@ public String getTitleText() {
-            init( versionRecordManager.getPathToLatest(), place, type, false, displayShowMoreVersions );
-            notification.fire( new NotificationEvent( CommonConstants.INSTANCE.ItemRestored() ) );
+            init(versionRecordManager.getPathToLatest(), place, type, false, displayShowMoreVersions);
+            notification.fire(new NotificationEvent(CommonConstants.INSTANCE.ItemRestored()));
         }
     }
 
     public void reload() {
         concurrentUpdateSessionInfo = null;
         refreshTitle();
-        baseView.showBusyIndicator( CommonConstants.INSTANCE.Loading() );
+        baseView.showBusyIndicator(CommonConstants.INSTANCE.Loading());
         loadContent();
-        changeTitleNotification.fire( new ChangeTitleWidgetEvent( place, getTitleText(), getTitle() ) );
+        changeTitleNotification.fire(new ChangeTitleWidgetEvent(place, getTitleText(), getTitle()));
     }
 
     private void disableMenus() {
-        disableMenuItem( COPY );
-        disableMenuItem( MenuItems.RENAME );
-        disableMenuItem( MenuItems.DELETE );
-        disableMenuItem( MenuItems.VALIDATE );
+        disableMenuItem(COPY);
+        disableMenuItem(MenuItems.RENAME);
+        disableMenuItem(MenuItems.DELETE);
+        disableMenuItem(MenuItems.VALIDATE);
     }
 
-    private void disableMenuItem( final MenuItems menuItem ) {
-        if ( menus.getItemsMap().containsKey( menuItem ) ) {
-            menus.getItemsMap().get( menuItem ).setEnabled( false );
+    private void disableMenuItem(final MenuItems menuItem) {
+        if (menus.getItemsMap().containsKey(menuItem)) {
+            menus.getItemsMap().get(menuItem).setEnabled(false);
         }
     }
 
-    /**
-     * If your editor has validation, overwrite this.
-     * @return The validation command
-     */
-    protected Command onValidate() {
+    protected boolean isValidationRunning = false;
+
+    protected Command getValidateCommand() {
+
         return new Command() {
             @Override
             public void execute() {
-                // Default is that nothing happens.
+                if (!isValidationRunning) {
+
+                    onBeforeValidate();
+
+                    onValidate(new Command() {
+                        @Override
+                        public void execute() {
+                            onAfterValidate();
+                        }
+                    });
+                }
             }
         };
     }
 
+    protected void onBeforeValidate() {
+        baseView.showBusyIndicator(CommonConstants.INSTANCE.Validating());
+        isValidationRunning = true;
+    }
+
+    protected void onAfterValidate() {
+        baseView.hideBusyIndicator();
+        isValidationRunning = false;
+    }
+
+    /**
+     * If your editor has validation, overwrite this.
+     * @return The validation command
+     */
+    protected void onValidate(final Command finished) {
+
```

Comment:
So shouldn't we throw an exception here to force override in child classes?

Reply:
Well, looking to other PRs of  the ensemble, probably we shouldn't.

Reply:
Yes, some need it, but not all. So this might be the best way.